src/radical/pilot/agent/radical-pilot-agent-multicore.py

```diff
@@ -5558,7 +5628,7 @@ def start_sub_agents(self):
 
                 with open (ls_name, 'w') as ls:
                     ls.write('#!/bin/sh\n\n')
-                    ls.write("%s\n" % cmd)
+                    ls.write("exec %s\n" % cmd)
```

Comment:
This is  also just tightening the process tree, and gives more direct cancellation.